### PR TITLE
Use `caption*` rather than `label*`

### PR DESCRIPTION
**Pull request by rowanc1**

### CODE REVIEW

File: src/nodes/figcaption.ts [hunk 1]
```diff
@@ -47,7 +47,10 @@ export const toMarkdown: MdFormatSerialize = (state, node) => {
 };
 
 export const toTex = createLatexStatement(
-  () => ({ command: 'caption', inline: true }),
+  (state) => ({
+    command: state.nextCaptionNumbered === false ? 'caption*' : 'caption',
```
Comment (rowanc1) [inline src/nodes/figcaption.ts:51]: This is the main change.

File: src/nodes/figure.ts [hunk 2]
```diff
@@ -79,21 +80,23 @@ export const toTex = createLatexStatement(
...
     const localId = state.options.localizeId?.(id ?? '') ?? id;
-    const star = numbered ? '' : '*';
     // TODO: Based on align attr
     // may have to modify string returned by state.renderContent(node);
     // https://tex.stackexchange.com/questions/91566/syntax-similar-to-centering-for-right-and-left
     state.write('\\centering');
     const captionFirst = node.firstChild?.type.name === nodeNames.figcaption;
     if (localId && captionFirst) {
       state.ensureNewLine();
-      state.write(`\\label${star}{${localId}}`);
```
Comment (rowanc1) [inline src/nodes/figure.ts:90]: I have removed `\label*{}` which I don't think is valid latex.

File: src/serialize/tex/utils.ts [hunk 3]
```diff
@@ -1,18 +1,16 @@
 /* eslint-disable no-param-reassign */
 import { Node } from 'prosemirror-model';
-import { TexStatementOptions, TexFormatSerialize, TexOptions, TexSerializerState } from '../types';
+import { TexStatementOptions, TexFormatSerialize, TexSerializerState } from '../types';
 
 export const INDENT = '  ';
 
 export function createLatexStatement(
-  opts: string | ((options: TexOptions, node: Node) => TexStatementOptions),
+  opts: string | ((state: TexSerializerState, node: Node) => TexStatementOptions),
```
Comment (rowanc1) [inline src/serialize/tex/utils.ts:8]: I updated this type, and the opts are on `state.options`.

**Review by rowanc1 — COMMENTED** [review]
Change review below!

File: src/nodes/figure.ts (hunk 2 above)
Comment (stevejpurves) [reply src/nodes/figure.ts:90]: Ah correct!

**Pull request merged**